modules/dnn/src/layers/nary_eltwise_layers.cpp

```diff
@@ -589,57 +596,99 @@ class NaryEltwiseLayerImpl CV_FINAL : public NaryEltwiseLayer
+                        ptr[i] = op(ptr1[i], ptr2[i], x3);
+                    }
+                } else {
+                    for(int i = r.start; i < r.end; i++, ptr1 += dp1, ptr2 += dp2, ptr3 += dp3, ptr += dp) {
                         *ptr = op(*ptr1, *ptr2, *ptr3);
+                    }
                 }
-            }
+            };
+            double nstripes = getNumThreads();
+            parallel_for_(Range(0, plane_size), worker, nstripes);
+        } else {
+            auto worker = [&](const Range &r) {
+                for (int plane_idx = r.start; plane_idx < r.end; plane_idx++) {
+                    const char* ptr1_ = data1;
+                    const char* ptr2_ = data2;
+                    const char* ptr3_ = data3;
+                    char* ptr_ = data;
+                    size_t idx = plane_idx;
+                    for (int k = ndims - 2; k >= 0; k--)
+                    {
+                        size_t next_idx = idx / shape[k];
+                        int i_k = (int)(idx - next_idx * shape[k]);
+                        ptr1_ += i_k * step1[k];
+                        ptr2_ += i_k * step2[k];
+                        ptr3_ += i_k * step3[k];
+                        ptr_ += i_k * step[k];
+                        idx = next_idx;
+                    }
+
+                    const T *ptr1 = (const T*)ptr1_;
+                    const T *ptr2 = (const T*)ptr2_;
+                    const T *ptr3 = (const T*)ptr3_;
+                    T* ptr = (T*)ptr_;
+                    if (dp1 == 1 && dp2 == 1 && dp3 == 1 && dp == 1) {
+                        for (int i = 0; i < plane_size; i++) {
+                            ptr[i] = op(ptr1[i], ptr2[i], ptr3[i]);
+                        }
+                    } else if (dp1 == 0 && dp2 == 1 && dp3 == 1 && dp == 1){
+                        T x1 = *ptr1;
+                        for (int i = 0; i < plane_size; i++) {
+                            ptr[i] = op(x1, ptr2[i], ptr3[i]);
+                        }
+                    } else if (dp1 == 1 && dp2 == 0 && dp3 == 1 && dp == 1){
+                        T x2 = *ptr2;
+                        for (int i = 0; i < plane_size; i++) {
+                            ptr[i] = op(ptr1[i], x2, ptr3[i]);
+                        }
+                    } else if (dp1 == 1 && dp2 == 1 && dp3 == 1 && dp == 1) {
+                        T x3 = *ptr3;
+                        for (int i = 0; i < plane_size; i++) {
+                            ptr[i] = op(ptr1[i], ptr2[i], x3);
+                        }
+                    } else {
+                        for(int i = 0; i < plane_size; i++, ptr1 += dp1, ptr2 += dp2, ptr3 += dp3, ptr += dp) {
+                            *ptr = op(*ptr1, *ptr2, *ptr3);
+                        }
+                    }
+                }
+            };
+            double nstripes = getNumThreads();
+            parallel_for_(Range(0, nplanes), worker, nstripes);
```

Comment:
> `nstripes = getNumThreads();`

This should not be used.
Already discussed several months ago - e.g. #23047

Reply:
Thank you for review but take it easy, this pr is still drafting. I still remember our discussion.

Reply:
Changed. Performance results are also updated.